### PR TITLE
Use filesystem for copying media when doing a network pull/push

### PR DESCRIPTION
**Pull request by petenelson**
### Description of the Change
Use filesystem for copying media when doing a network pull/push instead of `download_url()`

### Benefits
For network/multisite on the same server, there should be no need to make an HTTP request to download media when doing a pull or push. In some instances, due to DNS resolution, firewalls, HTTP authentication, etc, downloading media via a URL may fail. This update uses the filesystem to copy media rather than downloading it.

### Possible Drawbacks
None that I am aware of.

### Verification Process
Added some debugging code to save `dt_original_file_path` to post meta that is only updated when a file is copied successfully. Performed both multisite pushes and pulls to confirm the media is copied.

### Checklist:
- [X] I have read the [**CONTRIBUTING**](https://github.com/10up/distributor/blob/develop/CONTRIBUTING.md) document.
- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [X] All new and existing tests passed.

### Applicable Issues
#380 

### Changelog Entry
* Use filesystem for copying media when doing a network pull/push instead of `download_url()`

### CODE REVIEW

File: includes/classes/InternalConnections/NetworkSiteConnection.php [hunk 1]
```diff
@@ -144,7 +144,7 @@ public function push( $post_id, $args = array() ) {
 			 * @return {bool} If Distributor should push the post media.
 			 */
 			if ( apply_filters( 'dt_push_post_media', true, $new_post_id, $media, $post_id, $args, $this ) ) {
-				\Distributor\Utils\set_media( $new_post_id, $media );
+				\Distributor\Utils\set_media( $new_post_id, $media, [ 'use_filesystem' => true ] );
```
Comment (dinhtungdu) [inline includes/classes/InternalConnections/NetworkSiteConnection.php:147]: The PR works in my test. But I think we only use filesystem for network connection, so we can use `DT_IS_NETWORK` to detect multisite installations without introducing a new argument for `set_media` function.

Comment (dinhtungdu) [reply includes/classes/InternalConnections/NetworkSiteConnection.php:147]: @petenelson What do you think about my comment above? (Forgot to tag you 😅)

**Comment by jeffpaul** [comment]
As there's still a requested change on this PR, I'm going to move this to our next release so we can clear the 1.6 milestone and get to a release.

File: includes/classes/InternalConnections/NetworkSiteConnection.php (hunk 1 above)
Comment (dkotter) [reply includes/classes/InternalConnections/NetworkSiteConnection.php:147]: @dinhtungdu So I think the only issue using that constant here is that is only set to true if we are on multisite and if Distributor is network activated. So there's a chance someone could be using multisite but not have the plugin network activated, just activated on the individual sites they want to use. In that case, using that constant wouldn't work

**Review by dkotter — APPROVED** [review]

**Comment by dkotter** [comment]
Made some minor formatting changes and fixed the unit tests. This works well in my tests 👍

**Comment by iaingooey** [comment]
Hi There, Is there any chance of 2.1 coming out? just regarding the Featured images not pulling through to duplicate sites?

Cheers

**Pull request merged**